ms-engine/src/main/java/net/trajano/ms/engine/internal/VertxInputStream.java

```diff
@@ -0,0 +1,163 @@
+package net.trajano.ms.engine.internal;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.streams.ReadStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.Semaphore;
+
+/**
+ * <p></p>An input stream that wraps a Vert.X Read Buffer.  This does not do anything that performs blocks.</p>
+ * <p>This pauses() the stream once the buffer is read and resumes the stream when the buffer is fully read.</p>
+ */
+public class VertxInputStream extends InputStream {
+
+    private static final Buffer END_BUFFER = Symbol.newSymbol(Buffer.class);
+
+    private static final Buffer END_BUFFER_WITH_ERROR = Symbol.newSymbol(Buffer.class);
+
+    private static final Logger LOG = LoggerFactory.getLogger(VertxInputStream.class);
+    /**
+     * Read Stream being wrapped.
+     */
+    private final ReadStream<Buffer> readStream;
+    /**
+     * Semaphore to lock.  Permits is equivalent to available bytes.
+     */
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "LOG" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)